provisioning_client/adapters/hsm_client_riot.c

```diff
@@ -275,13 +274,13 @@ static int produce_device_cert(HSM_CLIENT_X509_INFO* riot_info, RIOT_ECC_SIGNATU
     {
         // Generating "root"-signed DeviceID certificate
         DERInitContext(&der_ctx, der_buffer, DER_MAX_TBS);
-        if (X509GetDeviceCertTBS(&der_ctx, &X509_DEVICE_TBS_DATA, &riot_info->device_id_pub) != 0)
+        if (X509GetDeviceCertTBS(&der_ctx, &X509_DEVICE_TBS_DATA, &riot_info->device_id_pub, (uint8_t*)&riot_info->ca_root_pub, sizeof(riot_info->ca_root_pub)) != 0)
         {
             LogError("Failure: X509GetDeviceCertTBS");
             result = MU_FAILURE;
         }
         // Sign the DeviceID Certificate's TBS region
-        else if ((status = RiotCrypt_Sign(&tbs_sig, der_ctx.Buffer, der_ctx.Position, (RIOT_ECC_PRIVATE*)eccRootPrivBytes)) != RIOT_SUCCESS)
+        else if ((status = RiotCrypt_Sign(&tbs_sig, der_ctx.Buffer, der_ctx.Position, &riot_info->device_id_priv)) != RIOT_SUCCESS)
```

Comment:
This block appears to be for the root-signed DeviceID case, but we're using the DeviceID key to sign the DeviceID cert.  This should be the root ca key.

Reply:
Fixed